### DISCUSSION TITLE
Some keybinds in Level5 layer doesn't work as expected

### DISCUSSION THREAD

**Discussion by bharathyes** [kovidgoyal/kitty#6510] [Q&A]
I use this keymapper(https://github.com/DreymaR/BigBagKbdTrixXKB) for a custom keyboard (colemak) with additional layer 5(extend layer) for navigation, backspaces, etc to be near the home row, and use the CapsLock as the level 5 modifier.

With Kitty some of the level 5 keybinds like backspace and escape dont work.

Eg: `CapsLock + o` should be backspace but rather emits `362u`.
likewise `Capslock + q` should be escape but does nothing.

When running `kitty +kitten show_key -m kitty` I got :

![kitty show_key out](https://github.com/kovidgoyal/kitty/assets/4404023/c66f4ce4-8df0-4338-b774-4ea58a1da8b8)

My sequence of keypress in above pic:
[1-2] my actual BS key, 
[3-6]`Caps + o` (o key is left of primary enter for my layout)
[7-8] actual escape key
[9-12] `Caps + q`

Initially, based on above output tried doing `map CAPS_LOCK no_op` and `map PAUSE no_op` but that didn't help. 

Correct me if that was incorrectly written.

I have also tried looking into the keyboard protocol in the kitty docs but haven't found how to proceed there. 

----

I don't mind much about other keybinds but is it possible to correct these two keybinds alone? 

Is this possible to fix or should I just get used to using the actual ESC and BS keys with kitten ?

**Comment by kovidgoyal** [comment]
Run with --debug-input and figure out what those keys appear as and map
that.

**Reply by bharathyes** [reply]
Thansk for the response. With the debug, for the same sequence of inputs mentioned previously, am getting : 

![image](https://github.com/kovidgoyal/kitty/assets/4404023/401c685e-8a8d-4990-a59e-9813962c8233)

The line for the `Caps+o` is BS shows : `Pause active_unknown_mods: Mod3 consumed_unknown_mods: Mod3 mods: numlock glfw_key: 57362 (PAUSE) xkb_key: 65299 (Pause) alternate_key: 57347 (BACKSPACE)`

and `Caps+q` is : `Caps_Lock active_unknown_mods: Mod3 consumed_unknown_mods: Mod3 mods: numlock glfw_key: 57358 (CAPS_LOCK) xkb_key: 65509 (Caps_Lock) alternate_key: 57344 (ESCAPE)`


I am stumped by what it means and what exactly I should use for map?

So far my attempts have been trying `map PAUSE no_op`, `map Pause no_op`, `map 57362 no_op`, `map 65299 no_op`, `map PAUSE BACKSPACE`. None of them seem to work. What would be the correct values to pass to map ?

-- 

Seems these two lines point to the issue, but unclear as to what it is or how to proceed:

`PRESS mods: numlock text: '' state: 0 ignoring as keyboard mode does not support encoding this event`

`PRESS mods: numlock text: '' state: 0 sent encoded key to child: ^[ [ 5 7 3 6 2 u `

**Comment by kovidgoyal** [comment]
mapping it to no_op is pointless. Its already a no_op you need to map it to send_text with whatever bytes you want it to send.

**Reply by bharathyes** [reply]
My reasoning was kitty might be overriding the default with its custom key-binds, so by giving no_op then the system wide key-bind will work.


On send_txt, these two worked to send the correct bytes like you said:

```
map ctrl+alt+o send_text all \x7f                                                   
map ctrl+alt+q send_text all \x1b 
```

First `alt` works only for left alt and the [modifier doc](https://sw.kovidgoyal.net/kitty/keyboard-protocol/#modifiers) doesn't mention one for level 5.

What would be the appropriate key combination to use in the config to map for my case above ?

Thanks for taking time to help BTW.
